### PR TITLE
MM-40415 : Feature flags are not reloaded client-side when server restarts

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
On server restart web socket event, client config are fetched again.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40415

#### Related Pull Requests
N/A

#### Screenshots
N/A

#### Release Note

```release-note
Feature flags are automatically refreshed when server undergoes restart
```

### CODE REVIEW

**Review by jespino — APPROVED** [review]
LGTM, good catch.

File: actions/websocket_actions.jsx [hunk 1]
```diff
@@ -182,6 +182,13 @@ export function unregisterPluginReconnectHandler(pluginId) {
     Reflect.deleteProperty(pluginReconnectHandlers, pluginId);
 }
 
+function restart() {
+    reconnect(false);
+
+    // We fetch the client config again on the server restart.
+    getClientConfig()(dispatch, getState);
```
Comment (hmhealey) [inline actions/websocket_actions.jsx:189]: Could you switch this around? These generally behave the same, but dispatching the result of `getClientConfig` works regardless of whether or not `getClientConfig` returns a thunk action or a plain redux action.
```suggestion
    dispatch(getClientConfig());
```

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: actions/websocket_actions.jsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply actions/websocket_actions.jsx:189]: yes thats makes sense

**Review by hmhealey — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
@jgilliam17 for this you might want to run the server and web app locally.

**Review by furqanmlk — APPROVED** [review]
thanks @M-ZubairAhmed 
Working as expected

1. Started the server `make run-server` with FeatureFlag `CallMobile=false`
2. Started WebApp and verified FeatureFlag is false
image
3. Changed FeatureFlag `CallMobile=true` in Server
4. ran `make restart-server`
5. In WebApp, verified FeatureFlag `CallMobile` has been set to `true`
image

**Pull request merged**